#15 add database

TandyN opened the pull request:
You guys can get a head start on reviewing this since it's a decent amount to review. I plan on adding additional commits to this PR.

These commits all refer to ticket #15

TandyN commented:
Starting at commit 'Created counter model', begins ticket #42

TandyN commented:
Starting at 'Created seeding script' , begins ticket #39

abhogle reviewed (changes requested):
1. In the **database/methods/counter.js** increment and decrement methods may have opportunity to explicitly handle errors before persisting reviews irrespective of increment being successful
2. In **server/server.js** app.get is handling review persistence. Here app.post may be more appropriate and inline with the actual purpose of the api call.

TandyN commented:
Counter methods themselves will not fail due to how mongoose works. Errors will be handled at the API level when a review fails to be added to the database.

I've corrected the main server.js file as I was using the line of code to test adding to the DB.

I've also added additional tests.

Please review the commit.

abhogle reviewed (approved):
Changes to main server.js look good. Keep in mind when you do add api handler to follow convention. No further changes are necesary at this point as long as counter.js is not swallowing errors. Please continue with merging into master.
